### PR TITLE
Add a command-line interface to the WML diff functionality

### PR DESCRIPTION
**Pull request by CelticMinstrel**
The diff functionality is there, so I figured it could be useful to expose it.

I would've just pushed straight to master instead of making a pull request, but I noticed output redirection doesn't seem to be working, so I decided to see if anyone had any idea why.

To be precise, the following command:

```
wesnoth.exe --diff test1.cfg test2.cfg > diff.cfg
```

outputs the diff to the console and not to diff.cfg. I tried both in VSCode's integrated terminal and in a standard Windows command prompt. I imagine it'll probably work as expected on Linux, but I'd like it to work on Windows too.

### CODE REVIEW

File: doc/man/wesnoth.6 [hunk 1]
```diff
@@ -104,6 +104,11 @@ information about command mode).
 .B --debug-lua
 enables some Lua debugging mechanisms
 .TP
+.BI -D,--diff \ left-file \  right-file
+diffs the two WML files; does not preprocess them first (to do that, run them through
+.B -p
+first. Outputs the diff as DiffWML on standard output.
```
Comment (jostephd) [inline doc/man/wesnoth.6]:
```suggestion
first). Outputs the diff as DiffWML on standard output.
```

**Comment by CelticMinstrel** [comment]
So, shadowm tells me that the output redirection issue is something that's unavoidable on Windows due to the way Wesnoth sets up its console redirection (unless we were to offer a separate Wesnoth executable for command-line use, maybe). Thus, I'm going to have to rethink this implementation, as the output file will need to be passed to Wesnoth rather than making use of shell redirection if this is to be useable on Windows.

I could either:

* Make the `--diff` and `--patch` flags require _three_ arguments rather than two.
* Add an additional flag to specify the output file.

Any thoughts?

As for a use-case... I think there could be a use-case but I've been unable to come up with a concrete example. This functionality _is_ used in the engine, but for typical WML files a text-based diff is probably more useful than this structural diff.

Maybe this could be useful for comparing different unit type definitions? A text-based diff would be sensitive to the order of keys, but if I understand correctly, the structural diff would not be. I would need to enable preprocessing in the diff function for that to be possible though...

**Comment by jostephd** [comment]
I think I prefer an additional flag to specify the output file. A --diff option that takes "left" and "right" arguments is intuitive; one that takes three arguments is not.

**Comment by CelticMinstrel** [comment]
I'd like to also make the `--output` flag work for the preprocess options, but it's a bit complicated; they actually output three files, so some overhaul of the preprocess options might be required for this to work.

**Comment by CelticMinstrel** [comment]
(What the heck is codacy on about...?)

**Comment by jostephd** [comment]
I think it might be confusing [our read() overload](https://github.com/wesnoth/wesnoth/blob/dbabf8d6461ab7869311e7ada9954511e914d974/src/serialization/parser.hpp#L26-L27) with [the read() syscall](http://pubs.opengroup.org/onlinepubs/9699919799/functions/read.html).

**Comment by CelticMinstrel** [comment]
Merged in d4c6f1e5f0b03e21320f421a73823cc56e8412bb.

**Pull request closed without merging**